File: dart/2145.dart
class Solution {
    int numberOfArrays(List<int> d, int l, int u) {
        int n = d.length;
        int c = l, s = l, e = l;
        for (int i = 0; i < n; i++) {
            c += d[i];
            if (s > c)
                s = c;
            if (e < c)
                e = c;
        }
        return max(0, s - l + u - e + 1);
    }
}

File: python3/2145.py
class Solution:
    def numberOfArrays(self, d: List[int], l: int, u: int) -> int:
        c = l
        r = [l]
        for i in d:
            c += i
            r.append(c)
        return max(0, min(r) - l + u - max(r) + 1)
